src/ccmain/tessedit.cpp

```diff
@@ -81,17 +83,13 @@ bool Tesseract::init_tesseract_lang_data(const std::string &arg0,
                                          bool set_only_non_debug_params, TessdataManager *mgr) {
   // Set the language data path prefix
   lang = !language.empty() ? language : "eng";
-  language_data_path_prefix = datadir;
-  language_data_path_prefix += lang;
-  language_data_path_prefix += ".";
+  std::filesystem::path tessdata_path = datadir / (lang + "." + kTrainedDataSuffix);
 
   // Initialize TessdataManager.
-  std::string tessdata_path = language_data_path_prefix + kTrainedDataSuffix;
-  if (!mgr->is_loaded() && !mgr->Init(tessdata_path.c_str())) {
-    tprintf("Error opening data file %s\n", tessdata_path.c_str());
-    tprintf(
+  if (!mgr->is_loaded() && !mgr->Init(tessdata_path.string().c_str())) {
+    tesserr << "Error opening data file " << tessdata_path.string() << '\n' <<
```

Comment:
```suggestion
    tesserr << "Error opening data file " << tessdata_path << '\n' <<
```

Reply:
Regarding the difference in path formatting between using `tesseract.exe a b -l c` with `tessdata_path.string()` versus `tessdata_path`:

- `tessdata_path.string()` produces a native OS path (e.g., on Windows):
```
Error opening data file X:\Projects\tesseract\c.traineddata
```
- `tessdata_path` produces an escaped path:
```
Error opening data file "X:\\Projects\\tesseract\\c.traineddata"
```
I believe it’s more user-friendly and intuitive to display the native OS path. Is there a specific reason we should use the escaped version instead? If not, I recommend sticking with the native path for error messages.